web/src/settings_org.ts

```diff
@@ -601,40 +653,52 @@ export function discard_stream_property_element_changes(elem, sub) {
             break;
         default:
             if (property_value !== undefined) {
-                settings_components.set_input_element_value($elem, property_value);
+                const validated_property_value = z
+                    .union([z.string(), z.number(), z.boolean()])
+                    .parse(property_value);
+                settings_components.set_input_element_value($elem, validated_property_value);
             } else {
                 blueslip.error("Element refers to unknown property", {property_name});
             }
     }
     update_dependent_subsettings(property_name);
 }
 
-export function discard_group_property_element_changes(elem, group) {
+export function discard_group_property_element_changes(elem: HTMLElement, group: UserGroup): void {
     const $elem = $(elem);
     const property_name = settings_components.extract_property_name($elem);
-    const property_value = settings_components.get_group_property_value(property_name, group);
+    const property_value = settings_components.get_group_property_value(
+        user_groups.user_group_schema.keyof().parse(property_name),
+        group,
+    );
 
     const group_widget_settings = [...settings_components.group_setting_widget_map.keys()];
     if (property_name === "can_mention_group") {
+        assert(typeof property_value === "number" || typeof property_value === "string");
         settings_components.set_dropdown_list_widget_setting_value(property_name, property_value);
     } else if (group_widget_settings.includes(property_name)) {
         const pill_widget = settings_components.get_group_setting_widget(property_name);
-        settings_components.set_group_setting_widget_value(pill_widget, property_value);
-    } else if (property_value !== undefined) {
```

Comment:
Removed these two lines because Sahil [said](https://github.com/zulip/zulip/pull/31940#discussion_r1799555034) this shouldn't be possible.